lib/vnet/osconfig_darwin.go

```diff
@@ -140,3 +140,35 @@ func vnetManagedResolverFiles() (map[string]struct{}, error) {
 	}
 	return matchingFiles, nil
 }
+
+// doWithDroppedRootPrivileges drops the privileges of the current process to those of the client
+// process that called the VNet daemon.
+func (c *osConfigurator) doWithDroppedRootPrivileges(ctx context.Context, fn func() error) (err error) {
+	rootEgid := os.Getegid()
+	rootEuid := os.Geteuid()
+
+	if err := syscall.Setegid(c.daemonClientCred.Egid); err != nil {
+		return trace.Wrap(err, "setting egid")
+	}
+	defer func() {
+		syscallErr := trace.Wrap(syscall.Setegid(rootEgid), "reverting egid")
```

Comment:
Does it make sense to do anything but panic if the OS prevents us from restoring the egid/euid that we had before? The whole process is basically unusable at that point right?

Reply:
True, true. I've never had a legitimate reason to use `panic` in any program that I've written, so I was hesitant to use it. But an error is technically recoverable, while here we just shouldn't continue execution at all.